src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const sepatareDate = date.split(fromFormat[3]);
+  const newDate = [];
+  const oldYearIndex = fromFormat.indexOf('YY');
+  const separator = toFormat[3];
+
+  if (sepatareDate[oldYearIndex] >= 30) {
+    sepatareDate[oldYearIndex] = '19' + sepatareDate[oldYearIndex];
+  }
+
+  if (sepatareDate[oldYearIndex] < 30) {
+    sepatareDate[oldYearIndex] = '20' + sepatareDate[oldYearIndex];
+  }
```

Comment:
The logic for determining the century of a two-digit year is incorrect. It should be based on whether `fromFormat` contains 'YY' or 'YYYY'. If 'YY', then the conversion logic should be applied. Also, the condition should be checked only if 'YY' is present in `fromFormat`.

Reply:
i corrected this